assets/src/components/ladder.tsx

```diff
@@ -185,42 +225,43 @@ const LadderTimepoint = ({
 
 const timepointStatusYFromTimepoints = (
   timepoints: Timepoint[],
-  timepointSpacingY: number
+  timepointSpacingY: TimepointSpacingYFunc
 ) => (
   timepointStatus: VehicleTimepointStatus | null,
   direction: VehicleDirection
-): number | null => {
+): TimepointSpacingYFunc => {
   if (timepointStatus) {
     const timepointIndex = timepoints.findIndex(
       timepoint => timepoint.id === timepointStatus.timepointId
     )
     if (timepointIndex !== -1) {
       const fractionDirection = direction === VehicleDirection.Up ? +1 : -1
-      return (
-        timepointSpacingY *
+      return (height: number) =>
+        timepointSpacingY(height) *
         (timepointIndex +
           timepointStatus.fractionUntilTimepoint * fractionDirection)
-      )
     }
   }
-  return null
+  return () => 0
 }
 
 const ScheduledLine = ({
   ladderVehicle,
   roadLineX,
   scheduledY,
+  height,
 }: {
   ladderVehicle: LadderVehicle
   roadLineX: number
   scheduledY: number
+  height: number
 }) => (
   <line
     className={`m-ladder__scheduled-line ${
       ladderVehicle.vehicle.scheduleAdherenceStatus
     }`}
     x1={ladderVehicle.x}
-    y1={ladderVehicle.y}
+    y1={ladderVehicle.y(height)}
```

Comment:
Would it be possible to use the `height` when calculating the `ladderVehicle` for the first time, instead of having to pass it in later? (This applies to a few different places.)

Reply:
No we don't know the height the first time, it is generated in an layout effect hook after this code is initially run.